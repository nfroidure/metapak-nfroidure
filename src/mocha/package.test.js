'use strict';

const packageTransformer = require('./package');

describe('mocha', () => {
  describe('Package transformer', () => {
    it('should work with some files', () => {
      expect(
        packageTransformer({
          scripts: {
            test: '',
          },
          metapak: {
            data: {
              testsFiles: 'src/*.js src/**/*.js',
            },
          },
        })
      ).toMatchSnapshot();
    });
  });
});
