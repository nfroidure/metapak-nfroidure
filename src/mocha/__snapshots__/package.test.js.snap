// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mocha Package transformer should work with babel config 1`] = `
{
  "devDependencies": {
    "coveralls": "^3.1.1",
    "mocha": "^10.0.0",
    "nyc": "^15.1.0",
  },
  "greenkeeper": {
    "ignore": [
      "mocha",
      "coveralls",
      "nyc",
    ],
  },
  "metapak": {
    "configs": [
      "babel",
      "mocha",
    ],
    "data": {
      "testsFiles": "src/*.js src/**/*.js",
    },
  },
  "nyc": {
    "exclude": [
      "src/*.js src/**/*.js",
    ],
  },
  "scripts": {
    "cover": "nyc npm test && nyc report --reporter=html --reporter=text",
    "coveralls": "nyc npm test && nyc report --reporter=text-lcov | coveralls && rm -rf ./coverage",
    "mocha": "mocha --require '@babel/register' src/*.js src/**/*.js",
    "test": "npm run mocha",
  },
}
`;

exports[`mocha Package transformer should work with some files 1`] = `
{
  "devDependencies": {
    "coveralls": "^3.1.1",
    "mocha": "^10.0.0",
    "nyc": "^15.1.0",
  },
  "greenkeeper": {
    "ignore": [
      "mocha",
      "coveralls",
      "nyc",
    ],
  },
  "metapak": {
    "configs": [
      "mocha",
    ],
    "data": {
      "testsFiles": "src/*.js src/**/*.js",
    },
  },
  "nyc": {
    "exclude": [
      "src/*.js src/**/*.js",
    ],
  },
  "scripts": {
    "cover": "nyc npm test && nyc report --reporter=html --reporter=text",
    "coveralls": "nyc npm test && nyc report --reporter=text-lcov | coveralls && rm -rf ./coverage",
    "mocha": "mocha src/*.js src/**/*.js",
    "test": "npm run mocha",
  },
}
`;
